UnityProject/Assets/Scripts/Systems/Voting/VotingManager.cs

```diff
@@ -102,18 +126,62 @@ public void TryInitiateRestartVote(GameObject instigator, NetworkConnection send
 		voteType = VoteType.RestartRound;
 		votePolicy = VotePolicy.MajorityRules;
 		voteInProgress = true;
-		RpcOpenVoteWindow("Vote restart initiated by", instigator.name, CountAmountString(), (30 - prevSecond).ToString());
+		RpcOpenVoteWindow("Vote restart initiated by", instigator.name, CountAmountString(), (30 - prevSecond).ToString(), yesNoList);
+		RpcVoteCallerDefault(sender);
+		Logger.Log($"Vote restart initiated by {instigator.name}", Category.Admin);
+	}
+
+	[Server]
+	public void TryInitiateNextGameModeVote(GameObject instigator, NetworkConnection sender = null)
+	{
+		if (voteInProgress || voteRestartSuccess) return;
+
+		if (isCooldownActive)
+		{
+			Chat.AddExamineMsgFromServer(instigator, $"Too soon to trigger a restart vote!");
+			return;
+		}
+
+		votes.Clear();
+		countTime = 0f;
+		prevSecond = 0;
+		voteType = VoteType.NextGameMode;
+		votePolicy = VotePolicy.MajorityRules;
+		voteInProgress = true;
+		RpcOpenVoteWindow("Vote restart initiated by", instigator.name, CountAmountString(), (30 - prevSecond).ToString(), GameModeList);
+		RpcVoteCallerDefault(sender);
+		Logger.Log($"Vote restart initiated by {instigator.name}", Category.Admin);
+	}
+
+	[Server]
+	public void TryInitiateNextMapVote(GameObject instigator, NetworkConnection sender = null)
+	{
+		if (voteInProgress || voteRestartSuccess) return;
+
+		if (isCooldownActive)
+		{
+			Chat.AddExamineMsgFromServer(instigator, $"Too soon to trigger a restart vote!");
+			return;
+		}
+
+		votes.Clear();
+		countTime = 0f;
+		prevSecond = 0;
+		voteType = VoteType.NextMap;
+		votePolicy = VotePolicy.MajorityRules;
+		voteInProgress = true;
+		RpcOpenVoteWindow("Vote restart initiated by", instigator.name, CountAmountString(), (30 - prevSecond).ToString(), MapList);
 		RpcVoteCallerDefault(sender);
 		Logger.Log($"Vote restart initiated by {instigator.name}", Category.Admin);
 	}
 
 	[Server]
-	public void RegisterVote(string userId, bool isFor)
+	public void RegisterVote(string userId, string isFor)
```

Comment:
Is `isFor` validated against the possible choices anywhere?

Reply:
the `isFor` is recorded and then at the end the game checks who voted for what.
the one string that has been detected the most is the winner.

Reply:
What is stopping a hacker from sending "trolled" as a vote?

Reply:
nothing... Until now! I've added a way for the server to check for 'sus' votes